src/main/java/roomescape/repository/reservation/ReservationJdbcRepository.java

```diff
@@ -39,20 +43,90 @@ public int deleteById(Long id) {
-                            resultSet.getString("name"),
-                            resultSet.getDate("date").toLocalDate(),
-                            new ReservationTime(
-                                    resultSet.getLong("time_id"),
-                                    resultSet.getTime("time_value").toLocalTime())
+                    ReservationTimeEntity reservationTimeEntity = new ReservationTimeEntity(
+                            resultSet.getLong("time_id"),
+                            resultSet.getTime("time_value").toLocalTime()
+                    );
+
+                    ThemeEntity themeEntity = new ThemeEntity(
+                            resultSet.getLong("theme_id"),
+                            resultSet.getString("theme_name"),
+                            resultSet.getString("theme_description"),
+                            resultSet.getString("theme_thumbnail"));
+
+                    return new ReservationEntity(
+                            resultSet.getLong("reservation_id"),
+                            resultSet.getString("reservation_name"),
+                            resultSet.getDate("reservation_date").toLocalDate(),
+                            reservationTimeEntity,
+                            themeEntity
                     );
-                    return reservation.withId(resultSet.getLong("reservation_id"));
                 }
         );
     }
+
+    @Override
+    public boolean existsByDateAndTime(LocalDate date, Long id) {
+        String sql = "select count(*) from reservation where date = ? and time_id = ?";
+        int count = jdbcTemplate.queryForObject(sql, Integer.class, date, id);
+        return count != 0;
+    }
+
+    @Override
+    public boolean existsByTimeId(Long timeId) {
+        String sql = "select count(*) from reservation where time_id = ?";
+        int count = jdbcTemplate.queryForObject(sql, Integer.class, timeId);
+        return count != 0;
+    }
+
+    @Override
+    public List<Long> findTimeIdsByDateAndTheme(LocalDate date, Long themeId) {
+        String sql = "SELECT time_id from reservation where date=? and theme_id=?";
+        List<Long> timeIds = jdbcTemplate.query(
+                sql,
+                (resultSet, rowNum) -> resultSet.getLong("time_id"),
+                date,
+                themeId);
+        return timeIds;
+    }
+
+    @Override
+    public List<Long> findTopThemesByReservationCountBetween(LocalDate startDate, LocalDate endDate) {
+        String sql = """
+                SELECT theme_id
+                FROM reservation
+                WHERE date BETWEEN ? AND ?
+                GROUP BY theme_id
+                ORDER BY COUNT(*) DESC
+                LIMIT 10
+                """;
+        List<Long> themeIds = jdbcTemplate.query(
+                sql,
+                (resultSet, rowNum) -> resultSet.getLong("theme_id"),
+                startDate,
+                endDate
+        );
+        return themeIds;
```

Comment:
```suggestion
        return jdbcTemplate.query(
                sql,
                (resultSet, rowNum) -> resultSet.getLong("theme_id"),
                startDate,
                endDate
        );
```
바로 return 해도 되겠네요. ㅎㅎ

Reply:
감사합니다!